File: Proga/Lab6/MyClasses/LowerString.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MyClasses
{
    public class LowerString : MyString
    {
        public LowerString(in string strIn)
        {
            int length = strIn.Length;
            this.str = new char[length];
            for (int i = 0; i < length; i++)
            {
                this.str[i] = strIn[i];
                if (!System.Char.IsLower(this.str[i]))
                    this.str[i] = System.Char.ToLower(this.str[i]);
            }
        }
        public override void MoveLetters()
        {
            int length = (this.str).Length;
            int amount = 1;
            char[] temp = new char[amount];
            for (int i = 0; i < amount; i++)
            {
                temp[i] = this.str[length - 1 - i];
            }
            for (int i = length - 1; i >= amount; i--)
            {
                this.str[i] = this.str[i - amount];
            }
            for (int i = 0; i < amount; i++)
            {
                this.str[amount - 1 - i] = temp[i];
            }
        }
        public override int GetLength()
        {
            return (this.str).Length;
        }
    }
}

File: Proga/Lab6/Lab6/LowerString.cpp
#include "LowerString.h"

// Public //

LowerString::LowerString(const char* strIn) 
{
    size_t length = strlen(strIn);
    this->str = new char[length + 1];
    for (size_t i = 0; i < length; i++)
    {
        this->str[i] = strIn[i];
        if (int(this->str[i]) >= 65 && int(this->str[i]) <= 90)
            this->str[i] = char(int(this->str[i]) + 32);
    }
    this->str[length] = '\0';
}

LowerString::LowerString(char* strIn)
{
    size_t length = strlen(strIn);
    this->str = new char[length + 1];
    for (size_t i = 0; i < length; i++)
    {
        this->str[i] = strIn[i];
        if (int(this->str[i]) >= 65 && int(this->str[i]) <= 90)
            this->str[i] = char(int(this->str[i]) + 32);
    }
    this->str[length] = '\0';
}

size_t LowerString::GetLength()
{
    return strlen(this->str);
}

void LowerString::MoveLetters()
{
    size_t length = strlen(this->str);
    size_t amount = 1;
    char* temp = new char[amount];
    for (size_t i = 0; i < amount; i++)
    {
        temp[i] = this->str[length - 1 - i];
    }
    for (size_t i = length - 1; i >= amount; i--)
    {
        this->str[i] = this->str[i - amount];
    }
    for (size_t i = 0; i < amount; i++)
    {
        this->str[amount - 1 - i] = temp[i];
    }
    delete[] temp;
}

// Private //